components/search/SearchBar.vue

```diff
@@ -56,108 +56,109 @@
   </div>
 </template>
 
-<script lang="ts">
-import {
-  Component,
-  Emit,
-  Prop,
-  Ref,
-  VModel,
-  Watch,
-  mixins,
-} from 'nuxt-property-decorator'
-import { SearchQuery } from './types'
-import PrefixMixin from '~/utils/mixins/prefixMixin'
-import KeyboardEventsMixin from '~/utils/mixins/keyboardEventsMixin'
+<script setup lang="ts">
 import { NeoAutocomplete } from '@kodadot1/brick'
 import { useCollectionSearch } from '@/components/search/utils/useCollectionSearch'
+import SearchSuggestion from '@/components/search/SearchSuggestion.vue'
+import { SearchQuery } from './types'
+import type { PropType } from 'vue'
 
-@Component({
-  components: {
-    SearchSuggestion: () => import('./SearchSuggestion.vue'),
-    NeoAutocomplete,
+const props = defineProps({
+  value: {
+    type: String,
+    required: true,
   },
+  query: Object as PropType<SearchQuery>,
 })
-export default class SearchBar extends mixins(
-  PrefixMixin,
-  KeyboardEventsMixin
-) {
-  @Prop({ type: Object, required: false }) public query!: SearchQuery
-  @VModel({ type: String }) name!: string
-  @Ref('searchRef') readonly searchRef
-  @Ref('searchSuggestionRef') readonly searchSuggestionRef
-  private enableSearchInCollection = true
-  public inputFocused = false
-
-  public created() {
-    this.initKeyboardEventHandler({
-      k: this.bindSearchEvents,
-    })
-    this.onSearchInCollectionModeChanged()
-  }
 
-  get isSearchInCollectionMode() {
-    return useCollectionSearch().isCollectionSearchMode.value
-  }
+const emits = defineEmits(['input', 'blur', 'enter', 'redirect'])
+const { $i18n } = useNuxtApp()
 
-  get placeholderContent() {
-    return this.inputFocused || this.isSearchInCollectionMode
-      ? ''
-      : this.$t('general.searchPlaceholder')
-  }
+const name = computed({
+  get: () => props.value,
+  set: (value) => {
+    emits('input', value)
+  },
+})
```

Comment:
`useVModel`?

Reply:
Hello @roiLeo ,

I think we can use useVModel here because based on the documentation ![image](https://github.com/kodadot/nft-gallery/assets/3810177/8e45860b-c9f4-4a2c-9f45-310a6f34792f)
It will emit `update:modelValue` which in our case is `update:value` but we want to emit `input` instead.

Please let me know how i can use this here as i might have overlooked other ways to use useVModel

Reply:
Alright noted

Reply:
@roiLeo  oh thank you for this didn't know we can do that. 4th argument is not documented here

https://github.com/vueuse/vueuse/blob/main/packages/core/useVModel/index.md

Reply:
@roiLeo I changed it to that. Thanks for the link